cmd/auth-provider-gcp/app/getcredentials_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func flagError(val string) error {
+	return fmt.Errorf("invalid value %q for authFlow (must be one of %q, %q, or %q)", val, gcrAuthFlow, dockerConfigAuthFlow, dockerConfigURLAuthFlow)
+}
+
+func TestValidateAuthFlow(t *testing.T) {
+	type FlagResult struct {
+		Flow  string
+		Error error
+	}
+	tests := []FlagResult{
+		{Flow: gcrAuthFlow, Error: nil},
+		{Flow: dockerConfigAuthFlow, Error: nil},
+		{Flow: dockerConfigURLAuthFlow, Error: nil},
+		{Flow: "bad-flow", Error: flagError("bad-flow")},
```

Comment:
Please add an empty Flow/string test.

Reply:
Also it looks like your code is case sensitive on these strings. I would add a test for this. (You are welcome to make it case insensitive first)

Reply:
Good ideas, done and done for both. I think case sensitivity for options is probably for the best here, so I left it in, but I did add a case sensitivity test so regressions won't be introduced.